google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/standard/AppEngineStandardWebIntegration.java

```diff
@@ -108,17 +108,18 @@ public void registerFrameworkInModel(FrameworkSupportModel model,
       AppEngineStandardFacet appEngineStandardFacet) {
   }
 
-  private void setupDeployRunConfiguration(@NotNull Project project) {
-    AppEngineCloudType serverType = ServerType.EP_NAME.findExtension(AppEngineCloudType.class);
+  private void setupDeployRunConfiguration(@NotNull Module module) {
+    AppEngineCloudType serverType =
+        ServerType.EP_NAME.findExtension(AppEngineCloudType.class);
     RemoteServer<AppEngineServerConfiguration> server
         = ContainerUtil.getFirstItem(RemoteServersManager.getInstance().getServers(serverType));
 
     DeployToServerConfigurationType configurationType
         = DeployToServerConfigurationTypesRegistrar.getDeployConfigurationType(serverType);
-    RunManager runManager = RunManager.getInstance(project);
+    RunManager runManager = RunManager.getInstance(module.getProject());
     ConfigurationFactoryEx factory = configurationType.getFactory();
-    RunnerAndConfigurationSettings settings
-        = runManager.createRunConfiguration(configurationType.getDisplayName(), factory);
+    RunnerAndConfigurationSettings settings = runManager.createRunConfiguration(
+        configurationType.getDisplayName() + " " + module.getName(), factory);
```

Comment:
The add framework support action assumes that a module exist. You can't trigger it if none exists.